instrumentation/spring/spring-rmi-4.0/javaagent/src/test/groovy/SpringRmiTest.groovy

```diff
@@ -0,0 +1,163 @@
+  }
+
+  def cleanupSpec() {
+    serverAppContext.close()
+    clientAppContext.close()
+  }
+
+  def "Client call creates spans"() {
+    given:
+    SpringRmiGreeter client = clientAppContext.getBean(SpringRmiGreeter)
+    when:
+    def response = runWithSpan("parent") { client.hello("Test Name") }
+    then:
+    response == "Hello Test Name"
+    assertTraces(1) {
+      trace(0, 3) {
+        span(0) {
+          name "parent"
+          kind SpanKind.INTERNAL
+          hasNoParent()
+        }
+        span(1) {
+          name "springrmi.app.SpringRmiGreeter/hello"
+          kind SpanKind.CLIENT
+          childOf span(0)
+          attributes {
+            "$SemanticAttributes.RPC_SYSTEM" "spring_rmi"
+            "$SemanticAttributes.RPC_SERVICE" "springrmi.app.SpringRmiGreeter"
+            "$SemanticAttributes.RPC_METHOD" "hello"
+          }
+        }
+        span(2) {
+          name "springrmi.app.SpringRmiGreeterImpl/hello"
+          kind SpanKind.SERVER
+          attributes {
+            "$SemanticAttributes.RPC_SYSTEM" "spring_rmi"
+            "$SemanticAttributes.RPC_SERVICE" "springrmi.app.SpringRmiGreeterImpl"
+            "$SemanticAttributes.RPC_METHOD" "hello"
+          }
+        }
+      }
+    }
+  }
+
+  def "Throws exception"() {
+    given:
+    SpringRmiGreeter client = clientAppContext.getBean(SpringRmiGreeter)
+    when:
+    runWithSpan("parent") { client.exceptional() }
+    then:
+    def error = thrown(IllegalStateException)
+    assertTraces(1) {
+      trace(0, 3) {
+        span(0) {
+          name "parent"
+          kind SpanKind.INTERNAL
+          status ERROR
+          hasNoParent()
+          event(0) {
+            eventName("$SemanticAttributes.EXCEPTION_EVENT_NAME")
+            attributes {
+              "$SemanticAttributes.EXCEPTION_TYPE" error.getClass().getCanonicalName()
+              "$SemanticAttributes.EXCEPTION_MESSAGE" error.getMessage()
+              "$SemanticAttributes.EXCEPTION_STACKTRACE" String
+            }
+          }
+        }
+        span(1) {
+          name "springrmi.app.SpringRmiGreeter/exceptional"
+          kind SpanKind.CLIENT
+          status ERROR
+          childOf span(0)
+          event(0) {
+            eventName("$SemanticAttributes.EXCEPTION_EVENT_NAME")
+            attributes {
+              "$SemanticAttributes.EXCEPTION_TYPE" error.getClass().getCanonicalName()
+              "$SemanticAttributes.EXCEPTION_MESSAGE" error.getMessage()
+              "$SemanticAttributes.EXCEPTION_STACKTRACE" String
+            }
+          }
+          attributes {
+            "$SemanticAttributes.RPC_SYSTEM" "spring_rmi"
+            "$SemanticAttributes.RPC_SERVICE" "springrmi.app.SpringRmiGreeter"
+            "$SemanticAttributes.RPC_METHOD" "exceptional"
+          }
+        }
+        span(2) {
+          name "springrmi.app.SpringRmiGreeterImpl/exceptional"
+          kind SpanKind.SERVER
+          status ERROR
```

Comment:
Is this span a `childOf span(1)`? The spring-rmi client instrumentation does not seem to implement context propagation; does the context get properly passed to the server?

Reply:
Yes, it is a `childOf span(1)`. For context propagation I'm using the Java RMI context propagation module, works perfectly fine using that.